crates/bevy_ui/src/render/mod.rs

```diff
@@ -788,6 +790,56 @@ pub fn extract_text_sections(
 
             end += 1;
         }
+
+        let Some(cursor) = maybe_cursor.filter(|cursor| !cursor.color.is_fully_transparent())
+        else {
+            continue;
+        };
+
+        let Some(glyph) = text_layout_info.glyphs.get(cursor.index) else {
+            continue;
+        };
+
+        let width = match cursor.width {
+            TextCursorWidth::All => glyph.size.x,
+            TextCursorWidth::Px(width) => width * target.scale_factor,
+        };
+
+        let height = computed_block.buffer().metrics().line_height * cursor.height;
```

Comment:
Line height may be unrelated to font size. I think this should be based on `max_ascent` + `max_descent`, but I'm not sure how difficult it would be to obtain that here.

Reply:
We can experiment I guess, we could even make a height enum that lets you base it off either font or line height. It's a less important part of this PR I think, we can adjust it later.

Reply:
Hm, it's hard for me to imagine line height being what people want, except when it's coincidentally a value based on font size. I've never personally seen an editor behave that way. I don't personally think that an a enum is going to be useful, but I don't mind punting this for later. People doing multi-line inputs with custom line heights are probably rare, and single-line people could probably work around the issue by using margins instead of line height.